AutoHub.DAL/Repositories/Repository.cs

```diff
@@ -61,19 +54,31 @@ public IEnumerable<T> AddRange(IEnumerable<T> newItems)
             return newItems;
         }
 
-        public bool Update(T item)
+        public T Update(T item)
         {
-            _context.Entry(item).State = EntityState.Modified;
-            return true;
+            return _context.Set<T>().Update(item).Entity;
         }
 
         public bool Delete(int id)
         {
             var toRemove = _context.Set<T>().Find(id);
-            if (toRemove == null) return false;
+            if (toRemove == null)
+                return false;
 
             _context.Remove(toRemove);
             return true;
         }
-    };
+
+        public IEnumerable<T> Include(params Expression<Func<T, object>>[] includes)
+        {
+            var query = _context.Set<T>().AsNoTracking();
+            return includes.Aggregate(query, (current, include) => current.Include(include)).ToList();
+        }
+
+        public IEnumerable<T> Include(Func<T, bool> predicate, params Expression<Func<T, object>>[] includes)
```

Comment:
This logic is the same as for GetAll but with Includes. You can call both these methods GetAll and it would be an overloaded version of GetAll method